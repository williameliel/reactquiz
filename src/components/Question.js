import React from 'react';

import Choices from './Choices';

function Question(props){

    if (!props.questions) {
        return false;
    }

    return (

      <div>
         <h2>{props.questions[props.current_step].question}</h2>
         <Choices 
           questionId={props.current_step}
           onChoiceSelect={props.onChoiceSelect}
           choices={props.questions[props.current_step].choices} />
      </div>

    );

}

export default Question;
