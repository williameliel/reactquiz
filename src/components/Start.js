import React from 'react';

const Start = ({title, description, handleStart}) => {

  return (
    <div>
      <h2>{title}</h2>
      <p>{description}</p>
      <button 
        className="btn btn-default" 
        onClick={handleStart}>Start</button>
    </div>
  );
};

export default Start;
