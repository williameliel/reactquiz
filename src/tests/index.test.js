import React from 'react';

import Start from '../components/Start';
import Question from '../components/Question';

import renderer from 'react-test-renderer';

it('Question renders correctly', () => {
  const tree = renderer.create(
    <Question />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});


it('Start renders correctly', () => {
  const tree = renderer.create(
    <Start />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
