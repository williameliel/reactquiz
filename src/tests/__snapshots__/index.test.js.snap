exports[`test Question renders correctly 1`] = `null`;

exports[`test Start renders correctly 1`] = `
<div>
  <h2 />
  <p />
  <button
    className="btn btn-default"
    onClick={undefined}>
    Start
  </button>
</div>
`;
